### PR TITLE
922 feature accordion component

### PR DESCRIPTION
**Pull request by sabato-galasso**
Fixes #

#### PR Checklist

- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes (if applicable).
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable).
- [x] I have added necessary documentation (if appropriate).

#### Short description of what this resolves:
Created new component Accordion with documentations and test

Screenshot 2023-01-23 alle 14 41 48
Screenshot 2023-01-23 alle 14 40 58

#### Changes proposed in this Pull Request:

- Added missing component AccordionItem

### CODE REVIEW

File: src/Accordion/AccordionItem.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React, { ElementType, FC } from 'react';
+import classNames from 'classnames';
+
+export interface AccordionItemProps {
+  /** Utilizzarlo in caso di utilizzo di componenti personalizzati */
+  tag?: ElementType;
+  /** Classi aggiuntive da usare per il componente Badge */
```
Comment (dej611) [inline src/Accordion/AccordionItem.tsx:7]:
```suggestion
  /** Classi aggiuntive da usare per il componente */
```

File: stories/Accordion/Accordion.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+  );
+};
+
+export const AccordionAnnidati = () => {
+  const [collapseOpen1, setCollapseOpen1] = useState(true);
+  const [collapseOpen2, setCollapseOpen2] = useState(false);
+  const [collapseOpen3, setCollapseOpen3] = useState(false);
+  const [nestedCollapseOpen1, setNestedCollapseOpen1] = useState(true);
+  const [nestedCollapseOpen2, setNestedCollapseOpen2] = useState(false);
+  const [nestedCollapseOpen3, setNestedCollapseOpen3] = useState(false);
```
Comment (dej611) [inline stories/Accordion/Accordion.stories.tsx:153]: nested collapsed should be mutually managed, see https://italia.github.io/bootstrap-italia/docs/componenti/accordion/#accordion-annidati

File: stories/Accordion/Accordion.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+    </Accordion>
+  );
+};
+
+HeaderAttivi.storyName = 'Header Attivi - sfondo primario';
+
+export const HeaderHover = () => {
+  const [collapseOpen1, setCollapseOpen1] = useState(true);
+  const [collapseOpen2, setCollapseOpen2] = useState(false);
+  const [collapseOpen3, setCollapseOpen3] = useState(false);
```
Comment (dej611) [inline stories/Accordion/Accordion.stories.tsx:295]: Should be mutually managed, see https://italia.github.io/bootstrap-italia/docs/componenti/accordion/#sfondo-primario

File: stories/Accordion/Accordion.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+    </Accordion>
+  );
+};
+
+HeaderHover.storyName = 'Header colorati on Hover - sfondo primario';
+
+export const IconaSinistra = () => {
+  const [collapseOpen1, setCollapseOpen1] = useState(true);
+  const [collapseOpen2, setCollapseOpen2] = useState(false);
+  const [collapseOpen3, setCollapseOpen3] = useState(false);
```
Comment (dej611) [inline stories/Accordion/Accordion.stories.tsx:343]: also this should be mutually managed: https://italia.github.io/bootstrap-italia/docs/componenti/accordion/#icona-a-sinistra

File: stories/Accordion/Accordion.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+# Accordion
+
+## Attiva/disattiva la visibilità di contenuti
+
+Per ottimizzare l’ingombro dei contenuti di una pagina si possono usare degli elementi richiudibili (in gergo definiti _collassabili_ o _collapse_), che possono essere attivati indipendentemente l’uno dall’altro oppure in modo esclusivo con l’attivazione di solo un blocco alla volta (in gergo definiti _fisarmoniche_ o _accordion_).
+
+### Come funziona
+
+Puoi usare un bottone per cambiare la visibilità dei contenuti:
```
Comment (dej611) [inline stories/Accordion/Accordion.stories.mdx:23]:
```suggestion
Per ottimizzare l’ingombro dei contenuti di una pagina a volte è necessario usare degli elementi richiudibili (in gergo definiti collassabili o collapse), che possono essere attivati indipendentemente l’uno dall’altro oppure in modo esclusivo con l’attivazione di solo un blocco alla volta (in gergo definiti fisarmoniche o accordion).
```

File: stories/Accordion/AccordionExample.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
```
Comment (dej611) [inline stories/Accordion/AccordionExample.tsx:1]: I think this example can be removed.

File: stories/Accordion/AccordionGroupsExample.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react';
```
Comment (dej611) [inline stories/Accordion/AccordionGroupsExample.tsx:1]: Shouldn't this be renamed to `CollapseGroupsExample` in the `Collapse` folder?

File: stories/Accordion/AccordionNestedExample.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
```
Comment (dej611) [inline stories/Accordion/AccordionNestedExample.tsx:1]: Where is this used?

File: stories/Accordion/AccordionExample.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    setCollapse(!collapse);
+  };
+  return (
+    <div>
+      <Button color='primary' onClick={toggle}>
+        Link with href
+      </Button>{' '}
+      <Button color='primary' onClick={toggle}>
+        Button with data-target
+      </Button>
```
[the remark this answers is not in the document]
Comment (sabato-galasso) [reply stories/Accordion/AccordionExample.tsx:17]: I deleted the file it is not used

File: stories/Accordion/AccordionExample.tsx (hunk 6 above)
Comment (sabato-galasso) [reply stories/Accordion/AccordionExample.tsx:1]: I deleted the file it is not used

File: stories/Accordion/AccordionGroupsExample.tsx (hunk 7 above)
Comment (sabato-galasso) [reply stories/Accordion/AccordionGroupsExample.tsx:1]: I deleted the file it is not used

**Pull request merged**